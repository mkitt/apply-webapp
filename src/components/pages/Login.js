// @flow
import React from 'react'
import glamorous from 'glamorous'
import type { RouterHistory } from 'react-router-dom'
import Button from '../elements/Button'
import ImageLink from '../elements/Link'
import Title from '../elements/Title'
import TuringMark from '../elements/TuringMark'

type Props = {
  history: RouterHistory,
}

export default (props: Props) => (
  <View>
    <ImageLink to="/">
      <TuringMark />
    </ImageLink>
    <Title mt={32}>Turing Apply</Title>
    <Button box centered primary mt={32} onClick={() => { props.history.push('/') }}>
      Authenticate with GitHub
    </Button>
  </View>
)

// -------------------------------------

const View = glamorous.div({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
})
