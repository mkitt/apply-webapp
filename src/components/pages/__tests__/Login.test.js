import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Login from '../Login'

describe('/login', () => {
  it('shows the correct snapshot tree for the /login page', () => {
    const component = (
      <MemoryRouter>
        <Login history={[]} />
      </MemoryRouter>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
