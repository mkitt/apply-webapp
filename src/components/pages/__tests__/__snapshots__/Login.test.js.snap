// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/login shows the correct snapshot tree for the /404 page 1`] = `
.glamor-3,
[data-glamor-3] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1,
[data-glamor-1] {
  font-size: 48px;
  font-weight: 200;
  color: #808080;
  margin-top: 32px;
}

.glamor-2,
[data-glamor-2] {
  text-align: center;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
  margin-top: 32px;
}

<div
  className="glamor-3"
>
  <a
    className="glamor-0"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="turing school mark"
      height={48}
      src="/turing-school-mark-256.png"
      width={48}
    />
  </a>
  <h1
    className="glamor-1"
  >
    Turing Apply
  </h1>
  <button
    className="glamor-2"
    onClick={[Function]}
    role="button"
  >
    Authenticate with GitHub
  </button>
</div>
`;

exports[`/login shows the correct snapshot tree for the /login page 1`] = `
.glamor-3,
[data-glamor-3] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-1,
[data-glamor-1] {
  font-size: 48px;
  font-weight: 200;
  color: #808080;
  margin-top: 32px;
}

.glamor-2,
[data-glamor-2] {
  text-align: center;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
  margin-top: 32px;
}

<div
  className="glamor-3"
>
  <a
    className="glamor-0"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="turing school mark"
      height={48}
      src="/turing-school-mark-256.png"
      width={48}
    />
  </a>
  <h1
    className="glamor-1"
  >
    Turing Apply
  </h1>
  <button
    className="glamor-2"
    onClick={[Function]}
    role="button"
  >
    Authenticate with GitHub
  </button>
</div>
`;
